src/lib.rs

```diff
@@ -201,6 +201,32 @@ pub mod container {
     }
 }
 
+/// Takes a reference to the first 16 bytes of the total bytes slice and convert it to an array for `peek_bytes` to use.
+/// Returns None if bytes's length is less than 16.
+pub fn take_hint_bytes(bytes: &[u8]) -> Option<&[u8; 16]> {
```

Comment:
I made this public, rather than working through the feature combination that would stop the compiler complaining about it being unused, because I felt it'd be a bit more maintainable - totally your call though.

Reply:
sorry I should have caught this earlier, can you make this private and then just do:
`#[allow(unused)]` to silence any warnings for any of the feature combos.
I'd like to have bare pub functions as small as possible, and etc.

Reply:
Makes sense 👍